adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -696,6 +690,63 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
   }
 
   /**
+   * This method gets the sort and partition map metadata from the header of the file
+   * give as input.
+   *
+   * @param filename the filename for the metadata
+   * @return a partition map if the data was written sorted, or an empty Seq if unsorted
+   */
+  private[rdd] def determineIsSortedAndExtractPartitionMap(filename: String): Seq[(ReferenceRegion, ReferenceRegion)] = {
+    // the sorted metadata is always stored with the sequence dictionary metadata
+    val path = new Path(filename + "/_seqdict.avro")
+    val fs = path.getFileSystem(sc.hadoopConfiguration)
+
+    // get an input stream
+    val is = fs.open(path).asInstanceOf[InputStream]
+
+    // set up avro for reading
+    val dr = new GenericDatumReader[GenericRecord]
+    val fr = new DataFileStream[GenericRecord](is, dr)
+
+    // parsing the json from the metadata header
+    // this unfortunately seems to be the only way to do this
+    // avro does not seem to support getting metadata fields out once you have the from the string
+    val metaDataMap = JSON.parseFull(fr.getMetaString("avro.schema")).get.asInstanceOf[Map[String, String]]
+
+    val maybePartitionMap = metaDataMap.get("partitionMap")
+    // we didn't write a partition map, which means this was not sorted at write
+    // or at least we didn't have information that it was sorted
+    if(maybePartitionMap.isEmpty) {
+      Seq()
+    } else {
+      // parsing the region information to rebuild the partition map
+      // when we wrote the partition map, we simply did a mkString on the
+      // Seq with commas separating each tuple of (ReferenceRegion, ReferenceRegion)
+      // The first split is breaking up the tuples. Each tuple starts with a
+      // "(" then a ReferenceRegion, so we are simply pulling out the tuples
+      // by using the start of each tuple as the indicator
```

Comment:
How about adding `ReferenceRegion` to the bdg-formats schemas and using Avro to write out the partition map?  I don't see why we should be writing some stuff out as Parquet, some as Avro, some as JSON, and some as Strings or byte-serialized JVM objects.